### PR TITLE
CLOUDP-89479: PodAntiAffinity sample CR 

### PR DESCRIPTION
**Pull request by FlorentinaSimlinger**
This PR adds a sample of a custom resource where podAntiAffinity is used.

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you signed all of your commits?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: config/samples/mongodb.com_v1_mongodbcommunity_cr_podantiaffinity.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      passwordSecretRef: # a reference to the secret that will be used to generate the user's password
+        name: my-user-password
+      roles:
+        - name: clusterAdmin
+          db: admin
+        - name: userAdminAnyDatabase
+          db: admin
+      scramCredentialsSecretName: my-scram
+  additionalMongodConfig:
+    storage.wiredTiger.engineConfig.journalCompressor: zlib
```
Comment (bznein) [inline config/samples/mongodb.com_v1_mongodbcommunity_cr_podantiaffinity.yaml:25]: [nit] can we make sure we make the sample files as minimal as possible? I know this line here is present in a few places but I think we included it once and it got copied to every other sample

File: config/samples/mongodb.com_v1_mongodbcommunity_cr_podantiaffinity.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+apiVersion: mongodbcommunity.mongodb.com/v1
+kind: MongoDBCommunity
+metadata:
+  name: example-mongodb
+spec:
+  members: 3
+  type: ReplicaSet
+  version: "4.2.6"
```
Comment (bznein) [inline config/samples/mongodb.com_v1_mongodbcommunity_cr_podantiaffinity.yaml:9]: [nit] can we put a newer version here? I think (outside the scope of this PR) we should periodically update this samples

Comment (FlorentinaSimlinger) [reply config/samples/mongodb.com_v1_mongodbcommunity_cr_podantiaffinity.yaml:9]: For the sake of consistency I suggest we do this in a follow up PR for all samples :)

Comment (bznein) [reply config/samples/mongodb.com_v1_mongodbcommunity_cr_podantiaffinity.yaml:9]: makes sense!

**Review by irajdeep — APPROVED** [review]

**Pull request merged**